crates/rattler_installs_packages/src/python_env/venv.rs

```diff
@@ -44,6 +47,16 @@ impl PythonLocation {
         match self {
             PythonLocation::System => system_python_executable(),
             PythonLocation::Custom(path) => Ok(path.clone()),
+            PythonLocation::CustomWithVersion(path, _) => Ok(path.clone()),
+        }
+    }
+
+    /// Get python version from executable
+    pub fn version(&self) -> Result<PythonInterpreterVersion, ParsePythonInterpreterVersionError> {
+        match self {
+            PythonLocation::System => PythonInterpreterVersion::from_system(),
+            PythonLocation::CustomWithVersion(_, version) => Ok(version.clone()),
+            PythonLocation::Custom(path) => PythonInterpreterVersion::from_path(path),
```

Comment:
just thinking out loud - do we also need to cache PythonInterpreterVersion::from_path? I'm thinking that the same python location will return the same version

Reply:
Yes that was indeed what I suggested!